Lib/test/test_urlparse.py

```diff
@@ -437,6 +437,12 @@ def test_urljoins(self):
         # issue 23703: don't duplicate filename
         self.checkJoin('a', 'b', 'b')
 
+        # issue 32779: clear the query string when joining with '?'
+        self.checkJoin('http://a/b/c?d=e', '?', 'http://a/b/c')
+        self.checkJoin('http://a/b/c?d=e', 'http:?', 'http://a/b/c')
```

Comment:
this is the test case that @orsenthil suggests is wrong as it doesn't match other languages which result in `http:?` as the output.

Reply:
Yes, that's correct. We need to remove this. Perhaps I will modify this and bring this PR to a close.